soc/arm/nxp_kinetis/k2x/soc.c

```diff
@@ -102,7 +102,7 @@ static ALWAYS_INLINE void clock_init(void)
 
 	CLOCK_SetSimConfig(&simConfig);
 
-#if CONFIG_USB_KINETIS || CONFIG_UDC_KINETIS
+#if CONFIG_USB_KINETIS || CONFIG_UDC_KINETIS || CONFIG_USB_UHC_NXP_KHCI
```

Comment:
Please add the SYSMPU block here if `CONFIG_ARM_MPU` and `CONFIG_SOC_MK22F12` are set, see my comment above

Reply:
Hi @danieldegrasse I add CONFIG_USB_UHC_NXP_KHCI like original CONFIG_UDC_KINETIS. The code snippet is used to enable the usb clock. Do you mean that I need to add `&& CONFIG_SOC_MK22F12` in this `#if`?

Reply:
Something like this:

```
#if CONFIG_USB_KINETIS || CONFIG_UDC_KINETIS || CONFIG_USB_UHC_NXP_KHCI
	CLOCK_EnableUsbfs0Clock(kCLOCK_UsbSrcPll0,
				CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
#if !defined(CONFIG_ARM_MPU) && defined(CONFIG_SOC_MK22F12 )
      /* Make sure SYSMPU is enabled */
      SYSMPU_Enable(SYSMPU, 0);
#endif
#endif
```

Reply:
Thanks Daniel, I have removed the codes of disable MPU. I use the same way as device in the include/zephyr/arch/arm/mpu/nxp_mpu.h file.